### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by ntzwilly**
## Description: 
- Refactor add and remove book features to persist changes in the server.
- Use middleware to make async requests to the API.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -44,18 +18,66 @@ export const removeBook = (id) => ({
...
+          progress: {
+            currentChapter: 'Introduction',
+            completed: '0',
+          },
         },
-      }];
+      ];
+    case SET_BOOKS: {
+      const saved = Object.entries(action.payload).map(([key, value]) => {
+        const [book] = value;
```
Comment (GzimAsani) [inline src/redux/books/books.js:66]: - [x] Please fix this error. And remember that the data must persist all changes in the server which the next reviewer will look after you fix this error
Screenshot from 2021-11-03 23-31-58

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @ntzwilly                             
Good job so far! Your code looks fine, good job :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the gitflow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [ ] _Check inline comments for more information :arrow_double_down:_ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (ntzwilly) [reply src/redux/books/books.js:66]: Thank you, this error is now fixed and the data persist in the server

**Review by dasileker — APPROVED** [review]
## **Hi @ntzwilly,** 

### **Optional**

- please remove the second readme file(old).

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**